components/popover/test/popover-mixin.test.js

```diff
@@ -151,4 +161,71 @@ describe('popover-mixin', () => {
 
 	});
 
+	describe('auto-focus', () => {
+
+		['open', 'toggleOpen'].forEach(method => {
+
+			it(`should focus on descendant when opened with ${method}`, async() => {
+				const elem = await fixture(html`<d2l-test-popover><button>shiny</button></d2l-test-popover>`);
+				await open(method, elem);
+				await nextFrame();
+				expect(document.activeElement).to.equal(elem.querySelector('button'));
+			});
+
+			it(`should focus on descendant when opened with ${method} and applyFocus is true`, async() => {
+				const elem = await fixture(html`<d2l-test-popover><button>shiny</button></d2l-test-popover>`);
+				await open(method, elem, true);
+				await nextFrame();
+				expect(document.activeElement).to.equal(elem.querySelector('button'));
+			});
+
+			it(`should not focus on descendant when opened with ${method} and applyFocus is false`, async() => {
+				const activeElement = document.activeElement;
+				const elem = await fixture(html`<d2l-test-popover><button>shiny</button></d2l-test-popover>`);
+				await open(method, elem, false);
+				await nextFrame();
+				expect(document.activeElement).to.equal(activeElement);
+			});
+
+			it(`should focus on popover when opened with ${method} and no focusable descendant`, async() => {
+				const elem = await fixture(html`<d2l-test-popover><span>shiny</span></d2l-test-popover>`);
+				await open(method, elem);
+				await nextFrame();
+				expect(document.activeElement).to.equal(elem);
+			});
+
+			it(`should not focus on descendant when opened with ${method} and no-auto-focus is true`, async() => {
+				const activeElement = document.activeElement;
+				const elem = await fixture(html`<d2l-test-popover no-auto-focus><button>shiny</button></d2l-test-popover>`);
+				await open(method, elem);
+				await nextFrame();
+				expect(document.activeElement).to.equal(activeElement);
+			});
+
+		});
+
+		['close', 'toggleOpen'].forEach(method => {
+
+			it(`should focus on opener when closed with ${method}`, async() => {
+				const elem = await fixture(html`
+					<div>
+						<button id="opener"></button>
+						<d2l-test-popover><button>shiny</button></d2l-test-popover>
```

Comment:
For this test could you just have `opened` set here?

Reply:
Not in this case. We need to render the fixture in order to apply focus to a fake opener before opening.